src/pages/TodoPage.tsx

```diff
@@ -1,21 +1,20 @@
 import TodoItem from "@/components/todo/TodoItem";
 import { Button } from "@/components/ui/button";
 import { CheckedState } from "@radix-ui/react-checkbox";
-import { useCallback, useMemo, useState } from "react";
-import { type ITodo } from "@/types/ITodo";
+import { useState } from "react";
+
+import type { ITodo } from "@/types/ITodo";
 
 export default function TodoPage() {
   const [todos, setTodos] = useState<ITodo[]>([]);
-  const todosCount = useMemo(() => todos.length, [todos]);
-  const nextId = useMemo(() => {
-    const next = todosCount + 1;
-    return next.toString();
-  }, [todosCount]);
 
-  const addTodo = useCallback(() => {
-    setTodos((prev) => [...prev, { done: false, id: nextId, name: "" }]);
-  }, [nextId]);
-  
+  const addTodo = () => {
+    setTodos((prev) => [
+      ...prev,
+      { done: false, id: (todos.length + 1).toString(), name: "" },
+    ]);
+  };
+
```

Comment:
so, my take on `useMemo()` and `useCallback()` isn't a final statement but rather a, a glance where i can imagine the truth. 

IF, every internal function to the component and every derived state (like `todo.length`) _should_ be wrapped into memoization functions (`useMemo()` and `useCallback()`) **don't you think that this would be auto-generated by React itself**

Consider these 2 functions as those ads from the 90s: take these pills and you'll loose the fat with NO EFFORT.

They are a scam.

please take a look to this post, (and the whole series realted to this one):

[how-to-use-memo-use-callback](https://www.developerway.com/posts/how-to-use-memo-use-callback)

it's really interesting ;)

***

im 101% sure that you shouldn't wrap `arr.length` into a `useMemo()`; 101% sure